code/game/objects/items/weapons/storage/backpack.dm

```diff
@@ -71,10 +71,19 @@
 		investigate_log("has become a singularity. Caused by [user.key]","singulo")
 		user << "<span class='danger'>The Bluespace interfaces of the two devices catastrophically malfunction!</span>"
 		qdel(W)
-		var/obj/singularity/singulo = new /obj/singularity (get_turf(src))
-		singulo.energy = 300 //should make it a bit bigger~
-		message_admins("[key_name_admin(user)] detonated a bag of holding")
-		log_game("[key_name(user)] detonated a bag of holding")
+		var/turf/T = get_turf(src)
+		var/area/A = T.loc
+		var/singulod = FALSE
+		if(istype(A, /area/arena))

```

Comment:
I do not approve of this special casing in this manner.

If we want to make specific areas immune to the sing, just make a flag or a var in /area and have the sing check it on New() and in process()